stl/inc/__msvc_int128.hpp

```diff
@@ -0,0 +1,1155 @@
+        }
+#else // ^^^ 128-bit intrinsics / no such intrinsics vvv
+        const auto __d             = _STD countl_zero(_Den._Word[1]);
+        const bool _Three_word_den = __d >= 32;
+        __d &= 31;
+        uint32_t __u[5]{
+            static_cast<uint32_t>(_Num._Word[0] << __d),
+            static_cast<uint32_t>(_Num._Word[0] >> (32 - __d)),
+            static_cast<uint32_t>(_Num._Word[1] << __d),
+            static_cast<uint32_t>(_Num._Word[1] >> (32 - __d)),
+            0,
+        };
+        if (__d != 0) {
+            __u[3] |= _Num._Word[0] >> (64 - __d);
+        }
+
+        uint32_t _Quotient[2];
+        if (_Three_word_den) {
+            // 4-digit by 3-digit base 2^32 division
+            uint32_t __v[3] = {
+                static_cast<uint32_t>(_Den._Word[0] << __d),
+                static_cast<uint32_t>(_Den._Word[0] >> (32 - __d)),
+                static_cast<uint32_t>(_Den._Word[1] << __d),
+            };
+            if (__d != 0) {
+                __v[2] |= _Den._Word[0] >> (64 - __d);
+            }
+
+            _Knuth_4_3_1_D(__u, __v, _Quotient);
+            _STL_INTERNAL_CHECK(__u[3] == 0);
+        } else {
+            // 4-digit by 4-digit base 2^32 division
+            uint32_t __v[4] = {
+                static_cast<uint32_t>(_Den._Word[0] << __d),
+                static_cast<uint32_t>(_Den._Word[0] >> (32 - __d)),
+                static_cast<uint32_t>(_Den._Word[1] << __d),
+                static_cast<uint32_t>(_Den._Word[1] >> (32 - __d)),
+            };
+            if (__d != 0) {
+                __v[2] |= _Den._Word[0] >> (64 - __d);
+            }
+            auto& __q = reinterpret_cast<uint32_t(&)[1]>(_Quotient);
+
+            _Knuth_4_3_1_D(__u, __v, __q);
+        }
+        _STL_INTERNAL_CHECK(__u[4] == 0);
+
+        _Num._Word[0] = static_cast<uint64_t>(__u[1]) << 32 | __u[0];
+        _Num._Word[1] = static_cast<uint64_t>(__u[3]) << 32 | __u[2];
+#endif // defined(_M_X64) && !defined(_M_ARM64EC)
+        _Num >>= __d;
+        return _Num;
+    }
+
+    constexpr _Uint128& operator%=(const uint32_t _Den) noexcept {
+        *this = *this % _Den;
+        return *this;
+    }
+    constexpr _Uint128& operator%=(const uint64_t _Den) noexcept {
+        *this = *this % _Den;
+        return *this;
+    }
+    constexpr _Uint128& operator%=(const _Uint128& _Den) noexcept {
+        *this = *this % _Den;
+        return *this;
+    }
+    template <integral _Ty>
+    friend constexpr _Ty& operator%=(_Ty& _Left, const _Uint128& _Right) noexcept {
+        _Left = static_cast<_Ty>(_Uint128{_Left} % _Right);
+        return _Left;
+    }
+
+    _NODISCARD friend constexpr _Uint128 operator&(const _Uint128& _Left, const _Uint128& _Right) noexcept {
+        return _Uint128{_Left._Word[0] & _Right._Word[0], _Left._Word[1] & _Right._Word[1]};
+    }
+
+    constexpr _Uint128& operator&=(const _Uint128& _That) noexcept {
+        _Word[0] &= _That._Word[0];

```

Comment:
Have you considered `_mm_and_si128` / `_mm_or_si128` / `_mm_xor_si128`, which is available for x64, as SSE2 is x64 baseline?

Reply:
Adding a bullet for this to #2520.